go/vt/vttablet/tabletserver/txthrottler/tx_throttler.go

```diff
@@ -356,14 +365,43 @@ func (ts *txThrottlerStateImpl) throttle() bool {
 	// Serialize calls to ts.throttle.Throttle()
 	ts.throttleMu.Lock()
 	defer ts.throttleMu.Unlock()
-	return ts.throttler.Throttle(0 /* threadId */) > 0
+
+	maxLag := atomic.LoadInt64(&ts.shardMaxLag)
+
+	return ts.throttler.Throttle(0 /* threadId */) > 0 &&
+		maxLag > ts.config.TxThrottlerConfig.TargetReplicationLagSec
```

Comment:
nit: this check can be reversed. 
```suggestion
	return maxLag > ts.config.TxThrottlerConfig.TargetReplicationLagSec &&
	        ts.throttler.Throttle(0 /* threadId */) > 0
```

Reply:
Makes sense.  I will do something similar in `txThrottler.Throttle()` to avoid calling the underlying logic if there were will be no throttling anyway due to query priority.

Reply:
Fixed